### PR TITLE
Fix subscription loading by policy name rest api

### PR DESCRIPTION
**Pull request by chamilaadhi**
- Fix subscription policy loading by name
- fix subscription policy per tenant
- add policy check for situation where policy is not in the map (due to tenant loading)

### CODE REVIEW

File: components/apimgt/org.wso2.carbon.apimgt.keymgt/src/main/java/org/wso2/carbon/apimgt/keymgt/handlers/AbstractKeyValidationHandler.java [hunk 1]
```diff
@@ -252,8 +254,26 @@ private APIKeyValidationInfoDTO validateSubscriptionDetails(APIKeyValidationInfo
 
                                 ApplicationPolicy appPolicy = datastore.getApplicationPolicyByName(app.getPolicy(),
                                         tenantId);
+                                if(appPolicy == null) {
```
Comment (mushthaq33) [inline components/apimgt/org.wso2.carbon.apimgt.keymgt/src/main/java/org/wso2/carbon/apimgt/keymgt/handlers/AbstractKeyValidationHandler.java:257]: Formatting issue

File: components/apimgt/org.wso2.carbon.apimgt.keymgt/src/main/java/org/wso2/carbon/apimgt/keymgt/handlers/AbstractKeyValidationHandler.java [hunk 2]
```diff
@@ -252,8 +254,26 @@ private APIKeyValidationInfoDTO validateSubscriptionDetails(APIKeyValidationInfo
...
+                                        appPolicy = new SubscriptionDataLoaderImpl()
+                                                .getApplicationPolicy(app.getPolicy(), apiTenantDomain);
+                                        datastore.addOrUpdateApplicationPolicy(appPolicy);
+                                    } catch (DataLoadingException e) {
+                                        log.error("Error while loading ApplicationPolicy");
+                                    }
+                                }
                                 SubscriptionPolicy subPolicy = datastore.getSubscriptionPolicyByName(sub.getPolicyId(),
                                         tenantId);
+                                if(subPolicy == null) {
```
Comment (mushthaq33) [inline components/apimgt/org.wso2.carbon.apimgt.keymgt/src/main/java/org/wso2/carbon/apimgt/keymgt/handlers/AbstractKeyValidationHandler.java:268]: Formatting issue

File: components/apimgt/org.wso2.carbon.apimgt.keymgt/src/main/java/org/wso2/carbon/apimgt/keymgt/service/APIKeyValidationService.java [hunk 3]
```diff
@@ -344,6 +347,19 @@ public APIKeyValidationInfoDTO validateKey(String context, String version, Strin
             if (storeImpl.isApiPoliciesInitialized()) {
                 log.debug("SubscriptionDataStore Initialized. Reading API Policies from SubscriptionDataStore");
                 apiPolicy = store.getApiPolicyByName(urlMapping.getThrottlingPolicy(), apiTenantId);
+                if(apiPolicy == null) {
```
Comment (mushthaq33) [inline components/apimgt/org.wso2.carbon.apimgt.keymgt/src/main/java/org/wso2/carbon/apimgt/keymgt/service/APIKeyValidationService.java:350]: Formatting issue

File: components/apimgt/org.wso2.carbon.apimgt.keymgt/src/main/java/org/wso2/carbon/apimgt/keymgt/handlers/AbstractKeyValidationHandler.java (hunk 2 above)
Comment (chamilaadhi) [reply components/apimgt/org.wso2.carbon.apimgt.keymgt/src/main/java/org/wso2/carbon/apimgt/keymgt/handlers/AbstractKeyValidationHandler.java]: done

File: components/apimgt/org.wso2.carbon.apimgt.keymgt/src/main/java/org/wso2/carbon/apimgt/keymgt/service/APIKeyValidationService.java (hunk 3 above)
Comment (chamilaadhi) [reply components/apimgt/org.wso2.carbon.apimgt.keymgt/src/main/java/org/wso2/carbon/apimgt/keymgt/service/APIKeyValidationService.java]: done

File: components/apimgt/org.wso2.carbon.apimgt.keymgt/src/main/java/org/wso2/carbon/apimgt/keymgt/handlers/AbstractKeyValidationHandler.java (hunk 1 above)
Comment (chamilaadhi) [reply components/apimgt/org.wso2.carbon.apimgt.keymgt/src/main/java/org/wso2/carbon/apimgt/keymgt/handlers/AbstractKeyValidationHandler.java]: done

**Pull request merged**